### PR TITLE
Remove redundant build, and pod install steps from benchmarks

### LINKED ISSUE #10697: Document how to create xcconfig file for running devicelab tests locally on iOS
When I run an iOS devicelab test locally, it tells me to specify a xcconfig. We should document on https://github.com/flutter/flutter/tree/master/dev/devicelab how to create that file and where the required information can be found.

/cc @yjbanov

### LINKED ISSUE #10729: Re-evaluate if benchmarks need to do a build before running
Currently, some benchmarks (I saw it on the `flutter_gallery_ios__transition_perf`) are doing a `flutter build` before doing a `flutter drive`. It seems like the build step is redundant because `flutter drive` will trigger another build (and since we don't have incremental builds, that one take as long as the first one). 

If we eliminate the first build, we would therefore also cut down the time it takes to execute benchmarks.

/cc @yjbanov

### PR DESCRIPTION
**Pull request by goderbauer**
Looks like these are no longer required as flutter will now handle these automatically. This should speed up our benchmarks.

Fixes https://github.com/flutter/flutter/issues/10729
Fixes https://github.com/flutter/flutter/issues/10697

### CODE REVIEW

**Comment by yjbanov** [comment]
+1 to removing redundant build and pod install. Unsure about provisioning profiles. Have you tried it with corporate-issues provisioning profiles (as opposed to personal profiles)?

**Comment by goderbauer** [comment]
Keeping the provisioning step, removing the extra build and pod install steps. PTAL.

File: dev/integration_tests/channels/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,10 +7,10 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		1498D2341E8E89220040F4C2 /* GeneratedPluginRegistrant.m in Sources */ = {isa = PBXBuildFile; fileRef = 1498D2331E8E89220040F4C2 /* GeneratedPluginRegistrant.m */; };
```
Comment (yjbanov) [inline dev/integration_tests/channels/ios/Runner.xcodeproj/project.pbxproj]: Not sure what's going on in this file.

**Review by yjbanov — APPROVED** [review]
LGTM

**Comment by xster** [comment]
Does this work if there were multiple certs on the machine prompting a choice? Or does it fall in with #10626 too? (e.g. --machine)

**Pull request merged**